### DISCUSSION TITLE
Keycloak idp login_hint forwarder

### DISCUSSION THREAD

**Discussion by CharlesCZ** [keycloak/keycloak#20019] [Q&A]
I need to forward username written in Keycloak login page to some idp e.g. github. So basically. Keycloak login page -> choose idp -> pass automatically username there I don't want to write username on idp side. I only want to write username on keycloak login page

**Comment by jbman** [comment]
In the IdP configuration you can enable to pass the login_hint from the authorization request to the target IdP
![image](https://user-images.githubusercontent.com/789496/235118030-17968784-100b-4cf5-a451-1af9d67bd9c1.png)

However the Keycloak login page won't pass the username input as login_hint, You still need your own client application which makes use of the login_hint URL parameter when directing the user to the Keycloak authorization endpoint. Maybe it could be achieved with a custom Keycloak theme.

**Reply by CharlesCZ** [reply]
Hi, thank's for response I'm using keycloak:21.0.2 but I cannot find it anywhere in Identity providers section
![Screenshot from 2023-04-27 16-50-02](https://user-images.githubusercontent.com/37272177/235122341-9a99dac2-7f59-4559-867b-eaa5664f26b1.png)

**Reply by jbman** [reply]
Maybe it is not available for specific social providers? The generic provider "Keycloak OIDC v1.0" has "Pass login_hint" at section "OpenID Connect settings" at the expandable "Advanced" sub-section.
Often it works to use the Keycloak OIDC v1.0 provider instead of a more specific one, but also could have some features which are then missing. 
Other option: I assume it's only the UI which is missing the setting, it may also be possible to use the Admin API to get it activated.

**Comment by CharlesCZ** [comment]
![Screenshot from 2023-04-28 12-23-37](https://user-images.githubusercontent.com/37272177/235122869-b26ffc4e-ed2c-4e97-aa21-3196c7cfb8ad.png)
